scipy/interpolate/fitpack2.py

```diff
@@ -1693,6 +1773,14 @@ def __init__(self, u, v, r, s=0., pole_continuity=False, pole_values=None,
         ider[1], ider[3] = pole_flat
 
         u, v = np.ravel(u), np.ravel(v)
+
+        if np.any(u <= 0.0) or np.any(np.pi <= u):
+            raise ValueError('u should be between (0, pi)')
```

Comment:
checks here are for open interval, while docstring says half-open

Reply:
I think that the docstring say `v` should be half open [-pi, pi], but `u` is open interval. But I agree that the evaluation code does not have good readability. So I improved it.

Reply:
I updated interval check.